_posts/2023-12-06-where-is-Strimzi-1.0.0.md

```diff
@@ -0,0 +1,86 @@
+Why does Strimzi still have only some _zero-dot-something_ versions?
+Where is Strimzi 1.0.0?
+So in this blog post, we will look at the history behind it and our latest plans for the 1.0.0 release.
+
+_This blog post is a personal opinion of the author and might not represent the view of the whole Strimzi community._
+
+<!--more-->
+
+### The history
+
+Strimzi is almost 6 years old.
+For a software project, 6 years is not a _short time_.
+When we initially started it, we started it _boldly_.
+Our first version was not called 0.0.1 but 0.1.0.
+The 0.1.0 release was not yet based on the operator pattern.
+That came only with the 0.2.0 release.
+0.1.0 was just a bunch of YAML files to make it easier to deploy Apache Kafka.
+
+Neither 0.1.0 nor 0.2.0 were of course production-ready.
+They were just the first steps on a long road.
+But with every release, we added more and more features, fixed bugs, and improved things ... and suddenly, we had our first production users and the first companies started to offer commercial products based on Strimzi.
+With more and more users, we started thinking about releasing Strimzi 1.0.0.
+
+But there was a thing on the horizon ... the _ZooKeeper-less Apache Kafka_ (also called _KRaft mode_).
+The [KIP-500](https://cwiki.apache.org/confluence/display/KAFKA/KIP-500%3A+Replace+ZooKeeper+with+a+Self-Managed+Metadata+Quorum) was published in August 2019.
+But already before we knew that there were plans to change the dependence on Apache ZooKeeper.
+Everyone knew that removing ZooKeeper from Kafka would not be a simple task.
+We knew that it would take a long time and mean many different changes.
+Changes to Apache Kafka, its Admin API and the way it is managed and configured.
+But also changes to Strimzi.
+For example, without ZooKeeper, the layout of our `Kafka` custom resource with its dedicated `.spec.kafka` and `.spec.zookeeper` sections does not make sense anymore.
+So we decided to not release Strimzi 1.0.0 and wait for the ZooKeeper removal to be finished.
+
+It turned out that the ZooKeeper removal from Apache Kafka took longer than we initially expected.
+In fact - it is still in progress - although it seems like it is slowly coming to an end and hopefully, next year will see the release of Apache Kafka 4.0 without ZooKeeper support.
+At the time of writing this blog post, Strimzi has 0.38.0 as its last release.
+That means 38 minor releases without releasing 1.0.0.
+
+So, looking back, the decision we took a long time ago was probably a mistake.
+If I had the chance to go back and change it, I would surely propose to do the 1.0.0 release already back then and now we would be talking only about _when to do the 2.0.0 release_.
+
+### Does it mean Strimzi is not production-ready?
+
+Absolutely not!
+We realize the 0.x version might seem scary and might put people off.
+However, the version number itself does not say much about the project maturity or about how good the project is.
```

Comment:
It may be good to say that we are NOT strictly following semantic versioning here, because [point 4](https://semver.org/#spec-item-4) explicitly says that 0.x is for initial development and anything may change at any time.

Reply:
It is a valid point, but it is not something I want to focus on in this blog post.